apps/openmw/mwrender/renderingmanager.cpp

```diff
@@ -251,6 +251,10 @@ namespace MWRender
         sceneRoot->setNodeMask(Mask_Scene);
         sceneRoot->setName("Scene Root");
 
+        mUniformRainIntensity = new osg::Uniform("rainIntensity",(float) 0.0);
+        
+        mRootNode->getOrCreateStateSet()->addUniform(mUniformRainIntensity);
```

Comment:
Could we look into setting the uniform on the water node rather than the root, so it doesn't clog the rest of the scene graph?

Reply:
I'm getting back to this right when I'm done with the weather particles. I have to enable the water ripples for snow so I will look into this too.